Feature: endpoints changes

violog opened the pull request:
Work not yet finished, but ready for reveiw

violog committed:
Drop balance activation endpoint ,disable PoH runner again

violog committed:
Filter events by their property to expire

violog committed:
Add endpoint for events configuration

violog committed:
Fix event types docs, change endpoint path to be JSON:API compliant

violog committed:
Small fix in event types docs and request validation

violog committed:
Change referral ID unit tests to use nullifier

Zaptoss commented on internal/data/events.go:
```diff
@@ -40,6 +40,7 @@ type ReopenableEvent struct {
 type EventsQ interface {
 	New() EventsQ
 	Insert(...Event) error
+	InsertOne(Event) (*Event, error)
```
We no longer need this method, the logic that used it now uses Update

violog committed:
Return a new state of referral code

violog committed:
Fix DB bugs with the new state of referral code

violog committed:
Add `is_verified` field in balance authorized response

violog committed:
Implement status in referral codes API model

violog commented:
Base reset to drop merged commits from older PRs

violog committed:
Fix bug with missing left join in referral code status

violog committed:
Resolve conflicts with main branch

# Conflicts:
#	internal/data/pg/referrals.go
#	internal/data/referrals.go
#	resources/model_resource_type.go

violog committed:
Paste active referral code status on balance creation, revert mocked auth

violog commented on resources/model_resource_type.go:
```diff
@@ -14,6 +14,7 @@ const (
 	CREATE_BALANCE   ResourceType = "create_balance"
 	UPDATE_BALANCE   ResourceType = "update_balance"
 	EVENT            ResourceType = "event"
+	EVENT_TYPE       ResourceType = "event_type"
```
generated code

violog commented:
Ready for review ✅

Zaptoss commented on internal/service/handlers/list_events.go:
```diff
@@ -26,6 +26,13 @@ func ListEvents(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if req.FilterHasExpiration != nil {
+		expirable := EventTypes(r).Names(func(ev evtypes.EventConfig) bool {
+			return ev.ExpiresAt == nil
+		})
+		req.FilterType = append(req.FilterType, expirable...)
+	}
```
If filter[has_expiration]=false anyway return only events which have expiration

violog committed:
Apply filter false has_expiration value in events
